packages/uikit/src/components/TokenLogo/TokenLogo.tsx

```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+import { HelpFilledIcon, HelpIcon } from "../Svg";
+
+export const TOKEN_LOGO_BAD_SRCS: { [imageSrc: string]: true } = {};
+
+export interface TokenLogoProps extends React.ImgHTMLAttributes<HTMLImageElement> {
+  srcs: string[];
+  useFilledIcon?: boolean;
+}
+
+/**
+ * Renders an image by sequentially trying a list of URIs, and then eventually a fallback triangle alert
+ */
+const TokenLogo: React.FC<React.PropsWithChildren<TokenLogoProps>> = ({ srcs, useFilledIcon, alt, ...rest }) => {
+  const [, refresh] = useState<number>(0);
+
+  const src: string | undefined = srcs.find((s) => !TOKEN_LOGO_BAD_SRCS[s]);
```

Comment:
I think `pakcages/uikit` shouldn't know `TOKEN_LOGO_BAD_SRCS`. This logic should be outside packages/uikit